examples/vault-iam-auth/user-data-auth-client.sh

```diff
@@ -90,10 +94,10 @@ login_output=$(retry \
 token=$(echo $login_output | jq -r .auth.client_token)
 
 # And use the token to perform operations on vault such as reading a secret
-response=$(curl \
-  -H "X-Vault-Token: $token" \
-  -X GET \
-  https://vault.service.consul:8200/v1/secret/example_gruntwork)
+# These is being retried because race conditions were causing this to come up null sometimes
```

Comment:
I could if I knew 😂 I mean, I just assumed the issue was due to a race condition since both the cluster & the client are being launched at the same time and sometimes the client would complain about things such as "role doesnt exist yet"

Reply:
That exact thing may be worth adding as a comment